onnxruntime/core/framework/execution_context.cc

```diff
@@ -0,0 +1,234 @@
+#include "core/framework/execution_context.h"
+#include "core/framework/execution_provider.h"
+#include "core/framework/execution_frame.h"
+#include "core/framework/bfc_arena.h"
+#include "core/framework/session_state.h"
+#include "core/common/spin_pause.h"
+
+namespace onnxruntime {
+
+ExecutionContext::ExecutionContext(const SessionState& sess_state,
+                                   int32_t num_streams,
+                                   gsl::span<const size_t> notification_owners,
+                                   gsl::span<const int> feed_mlvalue_idxs,
+                                   gsl::span<const OrtValue> feeds, gsl::span<const int> fetch_mlvalue_idxs,
+                                   std::vector<OrtValue>& fetches,
+                                   const InlinedHashMap<size_t, IExecutor::CustomAllocator>& fetch_allocators,
+                                   size_t num_barriers,
+                                   const logging::Logger& sess_logger,
+                                   const DeviceStreamCollection& device_stream_map,
+                                   bool single_thread_mode) : session_state_(&sess_state),
+                                                              frame_(feed_mlvalue_idxs,
+                                                                     feeds,
+                                                                     fetch_mlvalue_idxs,
+                                                                     fetches,
+                                                                     fetch_allocators,
+                                                                     sess_state,
+                                                                     device_stream_map.GetStreams()),
+                                                              logger_(&sess_logger),
+                                                              release_plan_(
+                                                                  new std::atomic_int[sess_state.GetExecutionPlan()->release_actions.size()]),
```

Comment:
Always create smart pointers on a separate line of code, never in a parameter list, so that a subtle resource leak won't occur due to certain parameter list allocation rules.
https://learn.microsoft.com/en-us/cpp/cpp/smart-pointers-modern-cpp?view=msvc-170